pull/github.go

```diff
@@ -829,6 +834,89 @@ func (ghc *GitHubContext) getCheckStatuses() (map[string]string, error) {
 	return statuses, nil
 }
 
+func (ghc *GitHubContext) LatestWorkflowRuns() (map[string]string, error) {
+	if ghc.workflowRuns != nil {
+		return ghc.workflowRuns, nil
+	}
+
+	workflowRuns, err := ghc.getWorkflowRuns()
+	if err != nil {
+		return nil, err
+	}
+
+	ghc.workflowRuns = workflowRuns
+
+	return workflowRuns, nil
+}
+
+func (ghc *GitHubContext) getWorkflowRuns() (map[string]string, error) {
+	// TODO: We could use the REST API to get the path, once
+	// https://github.com/google/go-github/commit/7665f317d3985efe6a7828e3af2751c41b3526b1
+	// is in a release.
+	var q struct {
+		Repository struct {
+			PullRequest struct {
+				Commits struct {
+					Edges []struct {
+						Node struct {
+							Commit struct {
+								CheckSuites struct {
+									Nodes []struct {
+										Conclusion  string
+										WorkflowRun struct {
+											File struct {
+												Path string
+											}
+										}
+									}
+									PageInfo *v4PageInfo
+								} `graphql:"checkSuites(first: 100, after: $checkSuiteCursor, filterBy: {appId: $appId})"`
+							}
+						}
+					}
+				} `graphql:"commits(last: 1)"`
+			} `graphql:"pullRequest(number: $number)"`
+		} `graphql:"repository(owner: $owner, name: $repo)"`
```

Comment:
If we stick with GraphQL, I think this query can be simplified by using the `object` query on `Repository` to directly get the head commit by SHA. This is also important for consistency - we want to make sure the commit we think we're evaluating is the one we actually get the check suites from.

Reply:
Cheers. I'm just checking on a real installation but I think the REST approach will work.